### PR TITLE
Fmu runner

### PR DESCRIPTION
**Pull request by nllong**
Add in the FMU runner using PyFMI. By default the FMU should use the parameters that were configured with the Model Exchange; however, the user can pass in the start, stop, and sim step if needed.

This also updates the Docker image to Ubuntu 20.04 and remove the jmodelica and ipython related code.

### CODE REVIEW

**Comment by nllong** [comment]
hey @vtnate! Can you merge this if all is good, then try to make named args work in the `spawn_docker.sh` file? Thanks!

**Comment by vtnate** [comment]
@nllong:
- I fixed type hints in fmu_runner.py, then my fix stopped working. I think fix 2.0 (what I have in there now) will hold.
- The optional arguments in spawn.py didn't work for me, so I made a few small changes.
- `ModelicaRunnerTest::test_compile_in_docker` and `CLIIntegrationTest ::test_cli_builds_sys_params` work locally. I haven't figured out why they doesn't in CI. No idea why `DistrictHeatingAndCoolingSystemsTest ::test_compile_district_heating_and_cooling_systems` fails, though it fails in the base branch (`spawn-optimica`) as well and it wasn't altered in this PR.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 name: CI
 
 on:
-  pull_request:
```
Comment (vtnate) [inline .github/workflows/ci.yml:4]: push encompasses pull_request. We were duplicating all our CI runs until I removed this line.

**Review by vtnate — APPROVED** [review]
🙌 We deleted jm_ipython.sh! Love this update, thanks for that beast of a Dockerfile

**Comment by nllong** [comment]
> @nllong:
> 
> * I fixed type hints in fmu_runner.py, then my fix stopped working. I think fix 2.0 (what I have in there now) will hold.
> * The optional arguments in spawn.py didn't work for me, so I made a few small changes.
> * `ModelicaRunnerTest::test_compile_in_docker` and `CLIIntegrationTest::test_cli_builds_sys_params` work locally. I haven't figured out why they doesn't in CI. No idea why `DistrictHeatingAndCoolingSystemsTest ::test_compile_district_heating_and_cooling_systems` fails, though it fails in the base branch (`spawn-optimica`) as well and it wasn't altered in this PR.

For posterity:

1) thanks for fixing my horrible typing implementation
2) Okay on fixing spawn.py
3) I can look at the CI (in your other PR). The test_compile won't work until we save the license and MACADDRESS to github secrets. I'll do that next.

**Pull request merged**